File: scripts/seed_admin.py
#!/usr/bin/env python3
"""Seed an admin user into the database."""
import asyncio
import sys
import os

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "backend"))

from app.database import connect_db, get_database
from app.models.user import new_user
from app.services.auth import hash_password


async def seed_admin(email: str, password: str, name: str = "Admin"):
    await connect_db()
    db = get_database()
    existing = await db.users.find_one({"email": email})
    if existing:
        print(f"User {email} already exists.")
        return
    user = new_user(email=email, password_hash=hash_password(password), role="admin", name=name)
    await db.users.insert_one(user)
    print(f"Admin user {email} created successfully.")


if __name__ == "__main__":
    email = "admin@yourwebsite.com" # Change this to your desired admin email before running
    password = "PASSWORD" # Change this to a secure password before running
    name = "MAD Admin"
    asyncio.run(seed_admin(email, password, name))

File: scripts/seed_admin.ts
import dotenv from 'dotenv';
import path from 'path';
dotenv.config({ path: path.resolve(__dirname, '../.env') });

import { getDb } from '../src/database';
import { newUser } from '../src/models/user';
import { hashPassword } from '../src/services/auth';

async function seedAdmin() {
  const email = process.env.ADMIN_SEED_EMAIL || 'admin@madphotography.com';
  const password = process.env.ADMIN_SEED_PASSWORD || 'changeme123';
  const name = process.env.ADMIN_SEED_NAME || 'Admin';

  const db = await getDb();
  const existing = await db.collection('users').findOne({ email });
  if (existing) {
    console.log(`Admin user ${email} already exists.`);
    process.exit(0);
  }

  const admin = newUser(email, hashPassword(password), 'admin', name);
  await db.collection('users').insertOne(admin);
  console.log(`Admin user created: ${email}`);
  process.exit(0);
}

seedAdmin().catch((err) => {
  console.error('Seed failed:', err);
  process.exit(1);
});
